canine/utils.py

```diff
@@ -355,6 +357,24 @@ def pandas_read_hdf5_buffered(key: str, buf: io.BufferedReader) -> pd.DataFrame:
         ) as store:
             return store[key]
 
+def base32(buf: bytes):
+    """
+    Convert a byte array into a base32 encoded string
+    """
+    table = np.array(list("abcdefghijklmnopqrstuvwxyz012345"))
+
+    bits = np.unpackbits(np.frombuffer(buf, dtype = np.uint8))
+    bits = np.pad(bits, (0, 5 - (len(bits) % 5)), constant_values = 0).reshape(-1, 5)
+    return "".join(table[np.ravel(bits@2**np.c_[4:-1:-1])])
```

Comment:
Wouldn't this be easier with `base64.b32encode()`?

Reply:
`b32encode` uses "/" in its alphabet, which is no bueno for filenames. Apparently older versions of `b32encode` let you specify the alphabet yourself, but newer versions do not.